oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/userinfo/DefaultOAuth2UserService.java

```diff
@@ -71,15 +74,14 @@
 
 	private static final String INVALID_USER_INFO_RESPONSE_ERROR_CODE = "invalid_user_info_response";
 
-	private static final ParameterizedTypeReference<Map<String, Object>> PARAMETERIZED_RESPONSE_TYPE = new ParameterizedTypeReference<Map<String, Object>>() {
-	};
-
 	private Converter<OAuth2UserRequest, RequestEntity<?>> requestEntityConverter = new OAuth2UserRequestEntityConverter();
```

Comment:
Should I improve `OAuth2UserRequestEntityConverter` as well, to allow for easier customizing by providing a setter for the Accept Header? 

Currently it is hardcoded to `APPLICATION_JSON` as well. 

This would allow the full jwt support by just implementing a custom `OidcUserInfoHttpMessageConverter` (application developer responsibility) and simple using `OAuth2UserRequestEntityConverter::setAcceptHeader("application/jwt")` instead of copying the source / implementing a custom solution.

Reply:
> Should I improve `OAuth2UserRequestEntityConverter` as well, to allow for easier customizing by providing a setter for the Accept Header?

Let's leave this for now. We might consider updating it but this should go into a separate issue/commit.